File: problems/algorithms/string-simple-substitution/js/response/simple-substitution.js
function substitution(message, plaintext, ciphertext) {
  plaintext = plaintext.split('');
  ciphertext = ciphertext.split('');
  const substitution = {};

  plaintext.forEach((char, index) => {
    substitution[char] = ciphertext[index];
  });

  return message
    .split('')
    .map(char => substitution[char] || char)
    .join('');
}

export { substitution };

File: problems/algorithms/string-simple-substitution/ruby/response/src/simple_substitution.rb
class Text
    def Text.substitution(message, plaintext, ciphertext)
        plaintext = plaintext.split('')
        ciphertext = ciphertext.split('')
        subs = {}

        plaintext.each_with_index { |char, index| subs[char] = ciphertext[index] }
        message = message.split('')
        message = message.map {|char| subs[char] || char}
        message = message.join('')

        return message
    end
end